app/src/main/java/com/example/coctailapp/ui/screens/main/content/cocktails/CocktailsContentViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package com.example.coctailapp.ui.screens.main.content.cocktails
+
+import android.content.Context
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import coil3.network.HttpException
+import com.example.coctailapp.R
+import com.example.coctailapp.model.CocktailsPreview
+import com.example.coctailapp.network.CocktailsApi
+import dagger.hilt.android.lifecycle.HiltViewModel
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import okio.IOException
+import javax.inject.Inject
+
+
+@HiltViewModel
+class CocktailsContentViewModel @Inject constructor(private val cocktailsApi: CocktailsApi,
+    @ApplicationContext val context : Context): ViewModel() {
+
+    private var _cocktailsList = MutableStateFlow(emptyList<CocktailsPreview>())
+    val cocktailsList = _cocktailsList.asStateFlow()
+
+
+    private var _dataFetchingState = MutableStateFlow<CocktailsFetchingEvent>(CocktailsFetchingEvent.LoadingEvent)
+    val dataFetchingEvent = _dataFetchingState.asStateFlow()
+
+    init {
+        getCocktailsByAlcoholic("Alcoholic")
+    }
+
+
+
+    fun getCocktailsByAlcoholic(alcoholic : String){
```

Comment:
This is the general comment on this method and its implementation. When testing, I encountered a bug where if the phone does not have the internet ( is in Airplane mode) the code goes to catch block but the ErrorEvent does not show any changes on the UI. The app is stuck with Loader on screen. Check this. Also, we should move the inital state LoadingEvent inside launch or event better inside try block